javaslang/src/main/java/javaslang/collection/Traversable.java

```diff
@@ -205,10 +201,9 @@
      * @return true, if this List contains all given elements, false otherwise.
      * @throws NullPointerException if {@code elements} is null
      */
-
     default boolean containsAll(Iterable<? extends T> elements) {
-        Objects.requireNonNull(elements, "elements is null");
-        return List.ofAll(elements).distinct().find(e -> !this.contains(e)).isEmpty();
+        HashSet<T> uniqueElements = HashSet.ofAll(elements);
+        return toSet().intersect(uniqueElements).size() == uniqueElements.size();
```

Comment:
this is another exciting change :)

E.g. for a `LinkedList` it had a complexity of `O(elements.size * size())`.
Now it's `O(elements.size + size())`. For sets it is reduced to the size of the smaller set.